be/src/util/hash_util.hpp

```diff
@@ -403,6 +403,17 @@ static size_t hash_bytes(void const* ptr, size_t const len) noexcept {
     return static_cast<size_t>(h);
 }
 
+// Xorshift is a pseudorandom number generator, which is an implementation of
+// linear-feedback shift registers (LFSRs). It generates next pseudorandom state from
+// the current state, which can be easily repurposed as hash functions.
+// The original paper: https://www.jstatsoft.org/article/view/v008i14
+static uint32_t xorshift32(uint32_t x) {
```

Comment:
Would this hash function change the data distribution of original hash function? Otherwise maybe introduce some data skew?

Reply:
The test result shows that there aren't data skew for all the scenarios. 
In other words, if it doesn't change the data distribution of the original hash function, how to solve the data skew when `num_fis` and `dop` aren't co-prime?

Reply:
> We have a seed set `Z` made up of m-tuples `(x_1, x_2, . . . , x_m)`, and a one-to-one function `f()` on `Z`. `z` is chosen uniformly and randomly from `Z`. Because `f` is 1-1 over `Z`, the random variable `f(z)` is itself uniform over `Z`.

I read some parts of the paper *Xorshift RNGs* just now. It says, if `x` is the uniform distribution and `xorshift32(x)` is a one-to-one function ,then `xorshift32(x)` is also a uniform distribution.